qa/tasks/mgr/dashboard/helper.py

```diff
@@ -277,25 +277,26 @@ def _request(cls, url, method, data=None, params=None, version=DEFAULT_VERSION):
         url = "{}{}".format(cls._base_uri, url)
         log.info("Request %s to %s", method, url)
         headers = {}
+        cookies = {}
         if cls._token:
             headers['Authorization'] = "Bearer {}".format(cls._token)
-
+            cookies['token'] = cls._token
```

Comment:
Here I'd choose either. Otherwise if we break either cookie or header authentication, we will never notice (as the other one will work).

What about adding a flag to the request to choose either cookie or header, and testing the header alone (as cookies are our new favorite) in a separate test?

Reply:
Yes it can be done, but the problem is we need to duplicate all the testing that we currently do using auth header  for cookies too, because we have to test each & every requests if it works by setting the cookie for token. I mean even if we use some helper function which does the same thing like login but with setting the cookies, how we will actually test in API tests. For e.g. in test_host.py qa task, it first calls a `setUpClass()` which basically sets up the cluster requirement including doing login(using auth header). So if we want to implement the same using cookies, don't we need to do `setUpClass` again!?

I think because for the test suites auth header & cookies work as a replacement, either can be used at one time. So if we want to test for both maybe we need to have separate test files with same test suites but using cookies?